Tella/Scenes/GDrive/ViewModel/GDriveOutboxViewModel.swift

```diff
@@ -0,0 +1,210 @@
+//
+//  GDriveOutboxViewModel.swift
+//  Tella
+//
+//  Created by gus valbuena on 7/10/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+import UIKit
+
+class GDriveOutboxViewModel: OutboxMainViewModel<GDriveServer> {
+    private let gDriveRepository: GDriveRepositoryProtocol
+    private var currentUploadCancellable: AnyCancellable?
+    private var uploadQueue: [ReportVaultFile] = []
+    var server: GDriveServer?
+    
+    init(mainAppModel: MainAppModel,
+         reportsViewModel : ReportsMainViewModel,
+         reportId : Int?,
+         repository: GDriveRepository) {
+        
+        self.gDriveRepository = repository
+        super.init(mainAppModel: mainAppModel, reportsViewModel: reportsViewModel, reportId: reportId)
+
+        if reportViewModel.status == .submissionScheduled {
+            self.submitReport()
+        } else {
+            self.pauseSubmission()
+        }
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(handleAppWillResignActive), name: UIApplication.willResignActiveNotification, object: nil)
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    @objc private func handleAppWillResignActive() {
+        if isSubmissionInProgress {
+            pauseSubmission()
+            updateReportStatus(reportStatus: .submissionPaused)
+        }
+    }
+    
+    private func getServer() {
+        self.server = mainAppModel.tellaData?.gDriveServers.value.first
+    }
+    
+    override func initVaultFile(reportId: Int?) {
+        getServer()
+        
+        if let reportId, let report = self.mainAppModel.tellaData?.getDriveReport(id: reportId) {
+            let vaultFileResult  = mainAppModel.vaultFilesManager?.getVaultFiles(ids: report.reportFiles?.compactMap{$0.fileId} ?? [])
+            
+            var files: [ReportVaultFile] = []
+            
+            report.reportFiles?.forEach({ reportFile in
+                if let vaultFile = vaultFileResult?.first(where: {reportFile.fileId == $0.id}) {
+                    let reportVaultFile = ReportVaultFile(reportFile: reportFile, vaultFile: vaultFile)
+                    files.append(reportVaultFile)
+                }
+            })
+            
+            self.reportViewModel = ReportViewModel(id: report.id,
```

Comment:
What do you think if you add the Init of ReportViewModel with Report and files 
`ReportViewModel(report:report, files:files)`